src/lavinmq/amqp/queue/queue.cr

```diff
@@ -414,6 +423,13 @@ module LavinMQ::AMQP
 
     def publish(msg : Message) : Bool
       return false if @deleted || @state.closed?
+      if d = @deduper
+        if d.duplicate?(msg)
+          @dedup_count += 1
+          return false
+        end
+        d.add(msg)
+      end
```

Comment:
No, we need to `return` from the method. Have some additional lines is good if it helps the reader IMO.

Reply:
don't understand.. my version also returns? 
yeah i guess.. i'm just allergic to nested if statements 😅

Reply:
No it doesn't :) the statement will be either false of Nil 
so if you do `my_value = d.duplicate?(msg) ? (@dedup_count += 1; false) : d.add(msg)` it will be `false` or `nil` but in order to return from the method you need `return false` after the `@dedup_count` 

we could extract it to a separate method like so: 

```
    # returns true if message is duplicated
    def handle_duplicated(msg : Message) : Bool
      return false unless d = @deduper
      if d.duplicate?(msg)
        @dedup_count += 1
        return true
      end
      d.add(msg)
      false
    end

    def publish(msg : Message) : Bool
      return false if @deleted || @state.closed?
      return false if duplicated_message(msg) 
     ....
```

but the method does so much different stuff, checking dups, adding to cache and increment counter so it's also not the best